### PR TITLE
intentionally harmful ecosystems

### PR DESCRIPTION
**Pull request by hardaker**
intentionally harmful ecosystems

### CODE REVIEW

File: draft-iab-protocol-maintenance.md [hunk 1]
```diff
@@ -372,6 +372,26 @@ by revising or extending it, makes the protocol better in the process.
 Tolerating unexpected input instead conceals problems, making it harder, if not
 impossible, to fix them later.
 
+## Intentionally harmful ecosystems
```
Comment (martinthomson) [inline draft-iab-protocol-maintenance.md:375]:
```suggestion
## Vendor Capture
```

File: draft-iab-protocol-maintenance.md [hunk 2]
```diff
@@ -372,6 +372,26 @@ by revising or extending it, makes the protocol better in the process.
...
+
+Vendors that have captured a significant portion of a protocol
+marketplace have been known to accidentally or intentionally create
+forks in defacto protocol standards.  These approaches, known as
+"embrace and extend" (or - worse - "embrace and deviate") have caused
+significant challenges in wide-scale interoperable deployment.
+Sometimes these have been rectified over time as a market dominant
+deviation has been brought back into interoperable standards
+specifications, such as the case with the deviation in the kerberos
+GSS API {{?GSS=RFC1964}}. [TODO: need to the RFC that microsoft's
```
Comment (martinthomson) [inline draft-iab-protocol-maintenance.md:385]: RFC 4121 ?

**Review by martinthomson — CHANGES_REQUESTED** [review]
We need to be careful here.

The GSS API example is probably a good one to demonstrate market capture, both because it is clearly abuse of market dominance and because the time has long passed.

However, I think that we should avoid dumping on a single vendor for abuses.  Microsoft were also at fault with HTML.  The use of "embrace and extend" (which was "embrace and extinguish") probably goes too far though.  We can point to the risks without laying it all at their feet so directly.

Are you aware of any good citations for SQL?

(I want to come back to this and turn this into two or three paragraphs.  One to lay out the market dominance risks and then some examples.)

File: draft-iab-protocol-maintenance.md (hunk 2 above)
Comment (hardaker) [reply draft-iab-protocol-maintenance.md:385]: Probably 4121 yes.

**Comment by hardaker** [comment]
That's a good point about not picking on a single vendor.  I wasn't trying to, and never actually spelled out who the vendor was in my proposed text, but it was [indeed] pointing toward a single one coincidentally.

Another issue to maybe highlight is the large gorillas that have blocked forward progress in interoperability to ensure vertical monopolies with their products.  Routing vendors, OS vendors, web browsers, etc have all played these games often even cooperating to ensure that all sides would not interoperate even when it might be best for the user.

**Comment by hardaker** [comment]
as for sql references -- nothing fantastic that I can think of.  Some not-great references:

https://en.wikibooks.org/wiki/SQL_Dialects_Reference/Introduction

https://www.quora.com/Why-are-there-so-many-types-of-SQL-What-is-the-difference-between-all-of-them?share=1

https://en.wikipedia.org/wiki/SQL#Reasons_for_incompatibility

**Review by DavidSchinazi — CHANGES_REQUESTED** [review]
I see the historical value of this paragraph, but it doesn't provide guidance. One way to summarize it is "some people were evil in the past and it sucked" but as a reader there's no real guidance. Advice along the lines of "don't be evil" won't be heeded by those who are doing harm intentionally. Given the risk of folks feeling singled out, I'd suggest avoiding this.

**Comment by martinthomson** [comment]
This PR extends the scope, albeit only a tiny bit, in a new direction.  That is, we aren't really talking about intentionally subverting the process by which a community defines what a protocol is, but more the adverse effects that come from neglect and casual abuse.  

That is, rather than engage with what is a rather tricky topic, we can call this one done.  (Not saying it isn't a problem, but maybe we can avoid making it *our* problem.)

**Pull request closed without merging**